core/src/main/java/org/infinispan/remoting/transport/jgroups/RspListFuture.java

```diff
@@ -0,0 +1,39 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import org.infinispan.remoting.responses.Response;
+import org.infinispan.util.concurrent.TimeoutException;
+import org.jgroups.blocks.GroupRequest;
+import org.jgroups.util.FutureListener;
+import org.jgroups.util.Rsp;
+import org.jgroups.util.RspList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+/**
+ * @author Dan Berindei
+ * @since 8.0
+ */
+class RspListFuture extends CompletableFuture<RspList<Response>> implements FutureListener<RspList<Response>> {
```

Comment:
Yes, JGroups gives me a single `GroupRequest` object. 

I could skip the JGroups' RpcDispatcher/MessageDispatcher building block and keep track of requests and responses myself, but I don't see a use case for it at the moment, and it would introduce a lot of complexity in our code that's now in JGroups.